### PR TITLE
[circle-execution-plan] Add tests to ExecutionPlanner for multi-outputs nodes

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This commit adds to ExecutionPlanner tests for multi-outputs nodes.

Merge it after #8109 

ONE-DCO-1.0-Signed-off-by: Artem Balyshev a.balyshev@partner.samsung.com

### CODE REVIEW

File: compiler/circle-execution-plan/src/ExecutionPlanner.test.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <loco.h>
+#include "ExecutionPlanner.h"
```
Comment (binarman) [inline compiler/circle-execution-plan/src/ExecutionPlanner.test.cpp:18]: nitpick:
- maybe swap these includes?
- do you need to include loco.h in the first place?

File: compiler/circle-execution-plan/src/ExecutionPlanner.test.cpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  split_out2->shape({1, 3});
+  split_out2->dtype(split->dtype());
+  split_out2->name("split_out2");
+
+  auto add = g->nodes()->create<luci::CircleAdd>();
+  add->name("add");
+  add->x(split_out1);
+  add->y(split_out2);
+  add->dtype(split_out1->dtype());
+  add->shape({1, 3});
```
Comment (binarman) [inline compiler/circle-execution-plan/src/ExecutionPlanner.test.cpp:75]: Note: if you are going to make larger test graphs, maybe it is easier to set only input shaped and dtypes, then run type and shape inference.

File: compiler/circle-execution-plan/src/ExecutionPlanner.test.cpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  // Create ExecutionPlanner
+  circle_planner::ExecutionPlanner execution_planner(g.get());
+  execution_planner.make_execution_plan();
+
+  // Check result
+  auto output1_exec_plan = luci::get_execution_plan(output1);
+  auto output2_exec_plan = luci::get_execution_plan(output2);
+
+  ASSERT_TRUE(output1_exec_plan.offsets().front() != output2_exec_plan.offsets().front());
+  ASSERT_TRUE(output1_exec_plan.offsets().front() + output1->dim(1).value() <=
```
Comment (binarman) [inline compiler/circle-execution-plan/src/ExecutionPlanner.test.cpp:171]: I don't get it. What does this check looking for?

P.s. this is also confusing: `offset` is measured in bytes, `dim(1).value` in number of elements

Comment (BalyshevArtem) [reply compiler/circle-execution-plan/src/ExecutionPlanner.test.cpp:171]: Yes, this is typo. I actually would like to check that offsets of the graph output nodes do not intersect. I will rewrite it

**Comment by BalyshevArtem** [comment]
@binarman,
I think, that it is better to close this pr now, in order to wait for all the pr associated with CircleExecutionPlan (#8128 #8099) to  merge and then write tests covering the entire ExecutionPlanner.
What do you think about this?

**Comment by binarman** [comment]
> I think, that it is better to close this pr now, in order to wait for all the pr associated with CircleExecutionPlan 

Do you plan to change planner interfaces dramatically?

If not, I don't see why we have to postpone this particular PR

File: compiler/circle-execution-plan/src/ExecutionPlanner.test.cpp [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+              output2_exec_plan.offsets().front() + output2->dim(1).value());
+
+  SUCCEED();
+}
```
Comment (seanshpark) [inline compiler/circle-execution-plan/src/ExecutionPlanner.test.cpp:175]: Please add same or more negative tests with `_NEG` ending at the name.
Our SE requires to provide more negative tests and currently numbers are at boundary.

@binarman , please check tests codes for this in `circle-execition-plan` and `luci-interpreter` tests which I don't fully review.
